lapis2/src/main/resources/application.properties

```diff
@@ -2,3 +2,12 @@ springdoc.default-produces-media-type=application/json
 springdoc.api-docs.path=/api-docs
 springdoc.swagger-ui.operationsSorter=alpha
 server.forward-headers-strategy=framework
+
+spring.cache.cache-names=siloQueryCache
+spring.cache.caffeine.spec=maximumSize=50000,expireAfterAccess=1h
```

Comment:
@chaoran-chen Do you have any experience from LAPIS 1 what reasonable values might be here? As a wild guess, I put 50,000 cached entries at max and a lifetime of 1 hours here.

Reply:
I don't have a good number for the number of entries but 50k sounds reasonable. Alternatively, is there a way to restrict the memory size? For LAPIS 1, the max size was set to 2 GB.

I don't think that we need an expiry time. As long as it's the same data version, the results will be the same.

Reply:
Thinking about it: the larget collection page (that I know of) sends around 2200 requests. Normal variant pages send around 20 relevant requests. So 5000 entries should probably do it here.

Reply:
Why only 5000? Or do you mean 50,000?

Don't forget that a collection is not static. You can filter by location and time range:

![image](https://github.com/GenSpectrum/LAPIS/assets/18666552/ce5968ab-8d77-40fa-9d27-ddda39ab3b90)

Reply:
Ah true. Ok, then I'll put it to 50k and no expiry time. Anyway, this config value can be overwritten when starting LAPIS.

Reply:
I did not find a way to make the eviction based on memory size. https://github.com/ben-manes/caffeine/wiki/Eviction#size-based

Maybe there are other cache implementations that support it. In principle, it should be easy to exchange. None of the implementation code is specific to the implementation.

Reply:
sounds good, then let's try with 50k and no expiry time